src/Authentication/Common.Authentication/Models/AzureContext.cs

```diff
@@ -12,24 +12,62 @@
 // limitations under the License.
 // ----------------------------------------------------------------------------------
 
+using Newtonsoft.Json;
+using System;
 namespace Microsoft.Azure.Common.Authentication.Models
 {
     /// <summary>
     /// Represents current Azure session context.
     /// </summary>
+    [Serializable]
     public class AzureContext
     {
-        public AzureContext(AzureSubscription subscription, AzureAccount account, AzureEnvironment environment)
+        /// <summary>
+        /// Creates new instance of AzureContext.
+        /// </summary>
+        /// <param name="subscription">The azure subscription object</param>
+        /// <param name="account">The azure account object</param>
+        /// <param name="environment">The azure environment object</param>
+        public AzureContext(AzureSubscription subscription, AzureAccount account, AzureEnvironment environment) :
+            this(subscription, account, environment, null)
```

Comment:
We should make sure that using this constructor doesn't lead to issues later in the code - the tenant id may be necessary in ARM, and may even be necessary in ASM